### PR TITLE
XmlLayout + JsonLayout - ExcludeProperties using StringComparer.OrdinalIgnoreCase

### PR DESCRIPTION
**Pull request by snakefoot**

### CODE REVIEW

**Comment by 304NotModified** [comment]
Nice. Maybe good idea to add unit tests?

**Comment by snakefoot** [comment]
Yes more unit-tests should be made, especially since the unit-tests from the Database-Target might move into its own repository.

After trying to use ValueTypeLayoutInfo more places, then I think it should be implemented differently. I don't like that one is required to inherit from the ValueTypeLayoutInfo.

File: appveyor.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 version: 4.7.0-{build}
 image:
-  - Visual Studio 2019
+  - Previous Visual Studio 2019
```
Comment (304NotModified) [inline appveyor.yml:3]: Why do we need this?

**Comment by snakefoot** [comment]
Because Microsoft broke Visual Studio in ver. 16.10. So rolling back to 16.9 until AppVeyor has updated their latest image with bugfix.


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


@304NotModified commented on this pull request.

________________________________

In appveyor.yml<https://github.com/NLog/NLog/pull/4470#discussion_r655777374>:

> @@ -1,6 +1,6 @@
 version: 4.7.0-{build}
 image:
-  - Visual Studio 2019
+  - Previous Visual Studio 2019


Why do we need this?

**Pull request merged**